src/views/service/BotServiceStatus.tsx

```diff
@@ -9,18 +9,22 @@ import * as BotServiceActions from '../../stores/service/actions';
 import { selectRequesting } from '../../stores/requesting/selectors';
 import IAppState from '../../services/store/IAppState';
 import { getBotServiceAsync, startBotServiceAsync, stopBotServiceAsync } from '../../stores/service/asyncActions';
+import useInterval from '../../hooks/useInterval';
 
 const ServicePage: React.FC = () => {
   const dispatch = useDispatch();
   const { botServices } = useSelector((state: IAppState) => state.botServiceStatus);
+  const [isPollingEnabled, setIsPollingEnabled] = useState(false)
   const isRequesting: boolean = useSelector((state: IAppState) =>
     selectRequesting(state, [BotServiceActions.REQUEST_BOT_SERVICE])
   );
+ 
+  useInterval(useCallback(() => dispatch(getBotServiceAsync()), [dispatch, getBotServiceAsync]), isPollingEnabled ? 3000 : null);
+
+  useEffect(()=>{
+    setIsPollingEnabled(true)
```

Comment:
**question**: I understand that we start the polling process as soon as we render this tab. But how is the polling process stopped? If it's not stopped, what happens if I switch to another tab and then switch back to this tab? Will I have multiple polling processes running?

Reply:
The useInterval hook calls clearInterval() when unmounting, so polling stops when you switching tabs